pkg/router/template/template_helper_test.go

```diff
@@ -805,3 +807,71 @@ func TestClipHAProxyTimeoutValue(t *testing.T) {
 		}
 	}
 }
+
+func TestGenerateHAProxyWhiteListFile(t *testing.T) {
+	workDir, err := ioutil.TempDir("", "haproxy")
+	if err != nil {
+		t.Fatal("Unable to create a temporary working directory")
+	}
+	defer os.RemoveAll(workDir)
```

Comment:
If you use `t.TempDir`, the resulting temporary directory will be automatically cleaned up when the test ends:
```suggestion
	workDir, err := t.TempDir("", "haproxy")
	if err != nil {
		t.Fatal("Unable to create a temporary working directory")
	}
```